pubsub/gcp/pubsub/pubsub.go

```diff
@@ -54,9 +54,10 @@ type GCPPubSub struct {
 	metadata *metadata
 	logger   logger.Logger
 
-	closed  atomic.Bool
-	closeCh chan struct{}
-	wg      sync.WaitGroup
+	closed     atomic.Bool
+	closeCh    chan struct{}
+	wg         sync.WaitGroup
+	topicCache map[string]bool
```

Comment:
See my other comment

```suggestion
	topicCache map[string]struct{}
```